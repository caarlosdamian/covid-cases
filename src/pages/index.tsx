export { DataGrid } from './DataGrid';
export { Login } from './Login';
