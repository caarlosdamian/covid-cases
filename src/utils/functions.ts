export const numberWithCommas = (number: number) =>
  number.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ',');
