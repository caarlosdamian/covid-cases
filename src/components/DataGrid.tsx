import { GridRowsProp } from '@mui/x-data-grid';
import { columns } from '../utils/tableColumns';
import { useApi } from '../hooks/useApi';
import { Table } from './table/Table';

export function DataGrid() {
  const { fetchData } = useApi();
  const rows: GridRowsProp = fetchData;

  return <Table columns={columns} rows={rows} />;
}
