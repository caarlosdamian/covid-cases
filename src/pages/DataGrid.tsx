import { GridRowsProp } from '@mui/x-data-grid';
import { columns } from '../utils/tableColumns';
import { useApi } from '../hooks/useApi';
import { Table } from '../components/table/Table';

export function DataGrid() {
  const { fetchData } = useApi();

  const rows: GridRowsProp = fetchData;

  return (
    <div>
      <Table data-testid="table" columns={columns} rows={rows} />
    </div>
  );
}
