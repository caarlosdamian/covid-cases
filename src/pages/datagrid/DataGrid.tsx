/* eslint-disable no-nested-ternary */
import { GridRowsProp } from '@mui/x-data-grid';
import { useTranslation } from 'react-i18next';
import { CircularProgress } from '@mui/material';
import { columns } from '../../utils/tableColumns';
import { useGetCovidDataQuery } from '../../hooks/useApi';
import { Table } from '../../components/table/Table';
import { Irow } from '../../interfaces';

export const DataGrid = () => {
  const { data, error, isLoading } = useGetCovidDataQuery();
  const rows: GridRowsProp = data as Irow[];

  const { t } = useTranslation('common');

  return (
    <div>
      {error ? (
        <span>{t('pages.datagrid.error')}</span>
      ) : isLoading ? (
        <CircularProgress />
      ) : data ? (
        <Table columns={columns} rows={rows} />
      ) : null}
    </div>
  );
};
