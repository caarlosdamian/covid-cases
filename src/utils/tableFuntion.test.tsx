/* eslint-disable no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import { describe, expect, it } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { numberWithCommas } from './functions';
import { columns } from './tableColumns';
import { DataGrid } from '../pages/DataGrid';

describe('Funtions working', () => {
  it('Should return the proper separation of numbers in a string', () => {
    expect(numberWithCommas(1000)).toBe('1,000');
  });

  it('Should return the proper separation of string numbers as a new string', () => {
    expect(numberWithCommas('1000')).toBe('1,000');
  });
});

// Each should be re-snapshotted if changed.
describe('All Snapshots Match', () => {
  it("Column's snapshot match", () => {
    expect(columns).toMatchSnapshot();
  });
});

// Tests around the DataGrid
describe('Components Render Correctly', () => {
  beforeEach(() => {
    render(<DataGrid />);
  });
  it('Renders Table without crashing', () => {
    expect(screen.getByTestId('table')).toBeDefined();
  });
  // Date should be shown while ID should not
  it('Renders columns Correctly', () => {
    expect(screen.getByText('Date')).toBeDefined();
    expect(screen.queryByText('ID')).toBeNull();
  });
  // ID can be turned on and off by the Columns
  it('Shows hidden Columns', async () => {
    fireEvent.click(screen.getByText('Columns'));
    fireEvent.click(screen.getByText('ID'));
    fireEvent.click(screen.getByText('Columns'));
    expect(await screen.findAllByText('ID')).toBeDefined();
  });
});
