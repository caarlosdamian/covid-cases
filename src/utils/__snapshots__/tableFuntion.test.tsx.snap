// Vitest Snapshot v1

exports[`All Snapshots Match > Column's snapshot match 1`] = `
[
  {
    "field": "date",
    "headerName": "ID",
    "width": 150,
  },
  {
    "field": "lastModified",
    "headerName": "Date",
    "type": "date",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "states",
    "headerName": "Infected States",
    "type": "number",
    "width": 150,
  },
  {
    "field": "positive",
    "headerName": "Positive",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "positiveIncrease",
    "headerName": "Positive Increase",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "negative",
    "headerName": "Negative",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "negativeIncrease",
    "headerName": "Negative Increase",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "pending",
    "headerName": "Pending",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "death",
    "headerName": "Deaths",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "deathIncrease",
    "headerName": "Death Increase",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "totalTestResults",
    "headerName": "Total Tests",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "totalTestResultsIncrease",
    "headerName": "Total Increase",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "hospitalized",
    "headerName": "Hospitalized",
    "type": "number",
    "valueFormatter": [Function],
    "width": 150,
  },
  {
    "field": "hospitalizedIncrease",
    "headerName": "Hospitalized Increase",
    "type": "number",
    "valueFormatter": [Function],
    "width": 170,
  },
]
`;
